genie-common-internal/src/main/java/com/netflix/genie/common/internal/util/RegexRuleSet.java

```diff
@@ -0,0 +1,207 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.common.internal.util;
+
+import com.google.common.collect.Lists;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class to match a string against an ordered set of regexes and obtain an accept/reject response.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+public final class RegexRuleSet {
+    private final Response defaultResponse;
+    private final List<Rule> rules;
+
+    private RegexRuleSet(
+        final LinkedList<Rule> rules,
```

Comment:
Any reason you take in a `LinkedList` instead of just a `List` as you store it in an `ArrayList` anyway?

Reply:
This should be `List`, i'll change it.
Started with the idea of having API to add rules with `addFirst` and `addLast` but then switched to builder so no longer necessary for it to be a linkedlist